huaweicloud/services/acceptance/dew/resource_huaweicloud_kms_grant_test.go

```diff
@@ -0,0 +1,139 @@
+package dew
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getKmsGrantResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getGrant: Query the KMS manual Grant
+	var (
+		getGrantHttpUrl = "v1.0/{project_id}/kms/list-grants"
+		getGrantProduct = "kms"
+	)
+	getGrantClient, err := cfg.NewServiceClient(getGrantProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating KMS Client: %s", err)
+	}
+
+	getGrantPath := getGrantClient.Endpoint + getGrantHttpUrl
+	getGrantPath = strings.ReplaceAll(getGrantPath, "{project_id}", getGrantClient.ProjectID)
+
+	getGrantOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		JSONBody: map[string]interface{}{
+			"key_id": state.Primary.Attributes["key_id"],
+			"limit":  100,
+		},
+	}
+	getGrantResp, err := getGrantClient.Request("POST", getGrantPath, &getGrantOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving KMS grant: %s", err)
+	}
+
+	grantResponseBody, err := utils.FlattenResponse(getGrantResp)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving KMS grant: %s", err)
+	}
+
+	searchPath := fmt.Sprintf("grants[?grant_id=='%s']|[0]", state.Primary.ID)
+	r, err := jmespath.Search(searchPath, grantResponseBody)
+	if err != nil || r == nil {
+		return nil, fmt.Errorf("error retrieving KMS grant: %s", err)
+	}
+
+	return r, nil
+}
+func TestAccKmsGrant_basic(t *testing.T) {
+	var obj interface{}
+
+	name := acceptance.RandomAccResourceName()
+	rName := "huaweicloud_kms_grant.test"
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getKmsGrantResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
```

Comment:
please add Admin check as the `huaweicloud_identity_user` was used in the test case

Reply:
added